rpc/client.go

```diff
@@ -184,44 +186,42 @@ func (c *Client) Call(ctx context.Context, addr, serviceMethod string, arg, repl
 	}
 
 	h := c.getClient(addr)
+	defer func() {
+		c.updateClientState(h, err, serviceMethod)
```

Comment:
If I'm reading correctly, the `h.cached == nil` check is not required in `updateClientState` because `h.Client()` will always be called before `c.updateClientState`, so the cached client will always be initialized. I think it'd be acceptable if we decided `updateClientState`'s contract is that the client must be initialized, because why reset before even using it, but I think it may be less fragile to just handle the uninitialized case. Or at least it should be documented.

Reply:
I came to the same conclusion, but was not 100% certain that I was missing something. I end up figuring that it was just as much overhead to make it safe as it was to document it, so I left it. Reasonable?